maestro-cli/src/main/java/maestro/cli/runner/MaestroCommandRunner.kt

```diff
@@ -74,16 +108,35 @@ object MaestroCommandRunner {
             onCommandStart = { _, command ->
                 logger.info("${command.description()} RUNNING")
                 commandStatuses[command] = CommandStatus.RUNNING
+                debugCommands[command] = CommandDebugMetadata(
+                    timestamp = System.currentTimeMillis(),
+                    status = CommandStatus.RUNNING
+                )
+
                 refreshUi()
             },
             onCommandComplete = { _, command ->
                 logger.info("${command.description()} COMPLETED")
                 commandStatuses[command] = CommandStatus.COMPLETED
+                debugCommands[command]?.let {
+                    it.status = CommandStatus.COMPLETED
+                    it.calculateDuration()
+                }
                 refreshUi()
             },
             onCommandFailed = { _, command, e ->
+                debugCommands[command]?.let {
+                    it.status = CommandStatus.FAILED
+                    it.calculateDuration()
+                    it.error = e
+                }
+
+                takeDebugScreenshot(CommandStatus.FAILED)
```

Comment:
Do we take screenshot only when command fails?

Reply:
yes